pkg/reconciler/dependencybuild/buildrecipeyaml.go

```diff
@@ -1,541 +1,192 @@
-
-          # FIXME: additionalArgs is added to args, but we need additionalArgs only; assume that we know the original tasks so that we can remove them
-          ADDITIONAL_ARGS=$(echo "$@" | sed 's/build publish \?//')
-          echo ADDITIONAL_ARGS="${ADDITIONAL_ARGS}"
-
-          if [ -n "$(params.ENFORCE_VERSION)" ]; then
-              gradle-manipulator --no-colour --info --stacktrace -l "${GRADLE_HOME}" $(params.GRADLE_MANIPULATOR_ARGS) -DversionOverride=$(params.ENFORCE_VERSION) "${ADDITIONAL_ARGS}" generateAlignmentMetadata || exit 1
-          else
-              gradle-manipulator --no-colour --info --stacktrace -l "${GRADLE_HOME}" $(params.GRADLE_MANIPULATOR_ARGS) "${ADDITIONAL_ARGS}" generateAlignmentMetadata || exit 1
-          fi
-
-          gradle -DAProxDeployUrl=file:$(workspaces.source.path)/hacbs-jvm-deployment-repo --info --stacktrace "$@" || exit 1
-
-          # fix-permissions-for-builder
-          chown 1001:1001 -R $(workspaces.source.path)
-      - name: deploy-and-check-for-contaminates
-        image: "registry.access.redhat.com/ubi8/ubi:8.5"
-        securityContext:
-          runAsUser: 0
-        script: |
-          tar -cvvzf $(workspaces.source.path)/hacbs-jvm-deployment-repo.tar.gz -C $(workspaces.source.path)/hacbs-jvm-deployment-repo .
-          curl --verbose --data-binary @$(workspaces.source.path)/hacbs-jvm-deployment-repo.tar.gz http://localhost:2000/deploy
+import (
+	_ "embed"
+	"github.com/redhat-appstudio/jvm-build-service/pkg/reconciler/artifactbuild"
+	"github.com/redhat-appstudio/jvm-build-service/pkg/reconciler/configmap"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"strings"
+)
 
-          curl --verbose --fail http://localhost:2000/deploy/result -o $(results.contaminants.path) || cat $(workspaces.maven-settings.path)/sidecar.log
-          cat $(workspaces.maven-settings.path)/sidecar.log
-        resources:
-          requests:
-            memory: "128Mi"
-            cpu: "10m"
-          limits:
-            memory: "512Mi"
-            cpu: "300m"
-`
+const (
+	WorkspaceBuildSettings = "build-settings"
+	WorkspaceSource        = "source"
 )
+
+//go:embed scripts/maven-settings.sh
+var mavenSettings string
+
+//go:embed scripts/gradle-settings.sh
+var gradleSettings string
+
+//go:embed scripts/maven-build.sh
+var mavenBuild string
+
+//go:embed scripts/gradle-build.sh
+var gradleBuild string
+
+//go:embed scripts/deploy.sh
+var deploy string
+
+func createPipelineSpec(maven bool, sidecarImage string, namespace string) *pipelinev1beta1.PipelineSpec {
+	var settings string
+	var build string
+	if maven {
+		settings = mavenSettings
+		build = mavenBuild
+	} else {
+		settings = gradleSettings
+		build = gradleBuild
+	}
+	zero := int64(0)
+	buildSetup := pipelinev1beta1.TaskSpec{
```

Comment:
I originally tried to initialize via golang struct formatting but kept running into presumably syntax errors that lead to runtime errors (vs. compile errors) instantiating objects, and then fell back to yaml based creation, and worrying about the yaml file sprawl.

I'm happy to see you were able to sort this out.

Reply:
this removes the needs around "verification" that I mentioned in our slack conversation, but said I would clarify within the PR.